dotCMS/src/main/java/com/dotcms/rest/RestEndPointIPUtil.java

```diff
@@ -0,0 +1,35 @@
+package com.dotcms.rest;
+
+import com.dotcms.util.HttpRequestDataUtil;
+import com.dotmarketing.util.UtilMethods;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Context;
+import java.util.Optional;
+
+/**
+ * Provide util methods to get remote or local IP from {@link HttpServletRequest}
+ */
+public class RestEndPointIPUtil {
```

Comment:
Codacy found an issue: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6190239)